### PR TITLE
[apps/examples/err_report_demo]: Fetch error report from console

### PR DESCRIPTION
**Pull request by dr-venkman**
Add application feature to print error report information. This feature is particularly useful when the device does not have an active network connection to a error reporting server.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [e48ed881c2cc7e64ee2fac991ee42d821c13a960] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [e48ed881c2cc7e64ee2fac991ee42d821c13a960] - Code Rule Check OK.

File: apps/examples/err_report_demo/err_report_main.c [hunk 1]
```diff
@@ -295,17 +295,35 @@ void error_report_process(int argc, char *argv[])
...
+			error_report_test(NULL);
 		} else {
-			error_report_records(argv[3]);
+			error_report_test(argv[3]);
 		}
-	} else if (strcmp(argv[2], "test") == 0) {
+	} else if (strcmp(argv[2], "report_send") == 0) {
 		if (argc < 4) {
-			error_report_test(NULL);
+			error_report_records(NULL);
```
Comment (jsdosa) [inline apps/examples/err_report_demo/err_report_main.c]: argc < 4
is that correct~?

File: apps/examples/err_report_demo/err_report_main.c [hunk 2]
```diff
@@ -295,17 +295,35 @@ void error_report_process(int argc, char *argv[])
...
 		} else {
-			error_report_test(argv[3]);
+			error_report_records(argv[3]);
+		}
+	} else if (strcmp(argv[2], "report_print")) {
```
Comment (SeonghoByeon) [inline apps/examples/err_report_demo/err_report_main.c]: In my opinion, list of commands might be 'test, send, print'. Please let me know if you any reason to attach prefix 'report_'?

File: apps/examples/err_report_demo/err_report_main.c [hunk 3]
```diff
@@ -295,17 +295,35 @@ void error_report_process(int argc, char *argv[])
...
+			int nentries = nbytes_read/sizeof(error_data_t);
+			for (i = 0; i < nentries; i++) {
+				printf("| %lu.%06lu | %04x | %d | %d | %04x |\n",
+						readptr->timestamp.tv_sec,
+						readptr->timestamp.tv_usec,
+						readptr->pc_value, readptr->module_id,
+						readptr->error_code, readptr->task_addr);
+				readptr++;
+			}
 		}
```
Comment (SeonghoByeon) [inline apps/examples/err_report_demo/err_report_main.c]: No error is reported here?

File: apps/examples/err_report_demo/err_report_main.c (hunk 2 above)
Comment (dr-venkman) [reply apps/examples/err_report_demo/err_report_main.c]: Thanks for the comment. Yes, the prefix is required to distinguish between printing an error report and printing trace calls (for callgraph) comprising different function invocations. In general, for each feature, a generic print method could be applied with a parameterized argument. This is however, beyond the scopr of this pull request. So for now, I mention report_print to make the context explicit.

File: apps/examples/err_report_demo/err_report_main.c (hunk 3 above)
Comment (dr-venkman) [reply apps/examples/err_report_demo/err_report_main.c]: No error, you are right. This part of the code simply prints out whatever errors have been logged up to this point. The error reporting module at present, does not report itself.

**Comment by seinfra** [comment]
Target : [e9dd18828193f9c8229e8bf4cddeafbdfdefc109] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [e9dd18828193f9c8229e8bf4cddeafbdfdefc109] - Code Rule Check (C++) OK.

File: apps/examples/err_report_demo/err_report_main.c (hunk 1 above)
Comment (dr-venkman) [reply apps/examples/err_report_demo/err_report_main.c]: Thank you for your comments. Yes, that is consistent with the fact that we pass argv[3] in the else block. Please let me know if I miss anything.

**Comment by seinfra** [comment]
Target : [544890b60767c69d3aa054f17ff93b73b033c119] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [544890b60767c69d3aa054f17ff93b73b033c119] - Code Rule Check OK.

File: apps/examples/err_report_demo/err_report_main.c [hunk 4]
```diff
@@ -295,17 +295,35 @@ void error_report_process(int argc, char *argv[])
...
 		if (argc < 4) {
-			error_report_test(NULL);
+			error_report_records(NULL);
 		} else {
-			error_report_test(argv[3]);
+			error_report_records(argv[3]);
+		}
+	} else if (strcmp(argv[2], "report_print")) {
+		int nbytes_read;
+		char readbuf[1024];
```
Comment (thapav) [inline apps/examples/err_report_demo/err_report_main.c]: You could use #define for defining buffer size

Comment (dr-venkman) [reply apps/examples/err_report_demo/err_report_main.c]: Thanks for the comment. I have modified as suggested.

**Comment by seinfra** [comment]
Target : [9e39b9213aa5dd64c05761c0f5759553307c7508] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [9e39b9213aa5dd64c05761c0f5759553307c7508] - Code Rule Check (C++) OK.

**Pull request merged**